packages/curve-ui-kit/src/shared/ui/DataTable.tsx

```diff
@@ -0,0 +1,155 @@
+import { useRef } from 'react'
+import Typography from '@mui/material/Typography'
+import TableHead from '@mui/material/TableHead'
+import TableCell from '@mui/material/TableCell'
+import TableBody from '@mui/material/TableBody'
+import Table from '@mui/material/Table'
+import TableRow from '@mui/material/TableRow'
+import { SizesAndSpaces } from '../../themes/design/1_sizes_spaces'
+import { useIntersectionObserver } from 'ui'
+import {
+  Cell,
+  Column,
+  ColumnDef,
+  flexRender,
+  getCoreRowModel,
+  getSortedRowModel,
+  Header,
+  Row,
+  SortingState,
+  useReactTable,
+} from '@tanstack/react-table'
+import { useSortFromQueryString } from '../../hooks/useSortFromQueryString'
+import { ArrowDownIcon } from '../icons/ArrowIcon'
+
+const { Sizing, Spacing, MinWidth } = SizesAndSpaces
+
+const getAlignment = <T extends any>({ columnDef }: Column<T>) => (columnDef.meta?.type == 'numeric' ? 'right' : 'left')
+
+const DataCell = <T extends unknown>({ cell }: { cell: Cell<T, unknown> }) => {
+  const column = cell.column
+  return (
+    <Typography
+      variant="tableCellMBold"
+      color="text.primary"
+      component="td"
+      sx={{
+        textAlign: getAlignment(column),
+        paddingInline: Spacing.sm,
+        paddingBlock: Spacing.md,
+        ...getExtraColumnPadding(column),
+      }}
+      data-testid={`data-table-cell-${column.id}`}
+    >
+      {flexRender(column.columnDef.cell, cell.getContext())}
+    </Typography>
+  )
+}
+
+const DataRow = <T extends unknown>({ row }: { row: Row<T> }) => {
+  const ref = useRef<HTMLTableRowElement>(null)
+  const entry = useIntersectionObserver(ref, { freezeOnceVisible: true })
+  return (
+    <TableRow
+      sx={{
+        marginBlock: 0,
+        height: Sizing['3xl'],
+        borderBottom: '1px solid',
+        borderColor: (t) => t.design.Layer[1].Outline,
+      }}
+      ref={ref}
+      // title={JSON.stringify(row.original)}
+      data-testid={`data-table-row-${row.id}`}
+    >
+      {entry?.isIntersecting && row.getVisibleCells().map((cell) => <DataCell key={cell.id} cell={cell} />)}
```

Comment:
This doesn't seem to work? When I decrease the width of my browser such that the cells are out of renderview, the dom elements are still there.